src/loan/protocol_tests.rs

```diff
@@ -783,6 +794,115 @@ mod tests {
+        fn coin_select(&mut self, amount: Amount, asset: AssetId) -> Result<Vec<Input>> {
+            let mut rng = thread_rng();
+
+            let (sk, pk) = make_keypair(&mut rng);
+            let (blinding_sk, blinding_pk) = make_keypair(&mut rng);
+            let address = Address::p2pkh(&pk, Some(blinding_pk.key), &AddressParams::LIQUID);
+
+            let spent_utxo_secrets = {
+                let abf = AssetBlindingFactor::new(&mut rng);
+                let vbf = ValueBlindingFactor::new(&mut rng);
+                let secrets = TxOutSecrets::new(asset, abf, amount.as_sat(), vbf);
+                let asset = Asset::new_confidential(SECP256K1, asset, abf);
+
+                [(asset, secrets)]
+            };
+
+            let (original_txout, _, _) = TxOut::new_last_confidential(
+                &mut rng,
+                SECP256K1,
+                amount.as_sat() + 10_000,
+                address,
+                asset,
+                spent_utxo_secrets
+                    .iter()
+                    .map(|(asset, secrets)| (*asset, secrets))
+                    .collect::<Vec<_>>()
+                    .as_slice(),
+                &[],
+            )?;
+
+            let txid = {
+                let mut bytes = [0u8; 32];
+                rng.fill_bytes(&mut bytes);
+
+                Txid::from_slice(&bytes)
+            }?;
+
+            let input = Input {
+                txin: { OutPoint { txid, vout: 0 } },
+                original_txout,
+                blinding_key: blinding_sk,
+            };
+
+            self.utxos.push((input.clone(), sk, pk));
+
+            Ok(vec![input])
+        }
+
+        fn sign_inputs(&self, tx: Transaction) -> Transaction {
+            let mut tx_to_sign = tx;
+            // first try to find out which utxos we know
+            let known_inputs = tx_to_sign.clone().input.into_iter().filter_map(|txin| {
+                if let Some((input, sk, pk)) = self.utxos.iter().find(
+                    |(
+                        Input {
+                            txin: OutPoint { txid, .. },
+                            ..
+                        },
+                        _,
+                        _,
+                    )| *txid == txin.previous_output.txid,
+                ) {
+                    Some((txin, input.original_txout.value, sk, pk))
+                } else {
+                    None
+                }
+            });
+
+            known_inputs.into_iter().for_each(|(txin, value, sk, pk)| {
+                let hash = elements::hashes::hash160::Hash::hash(&pk.serialize());
+                let script = Builder::new()
+                    .push_opcode(opcodes::all::OP_DUP)
+                    .push_opcode(opcodes::all::OP_HASH160)
+                    .push_slice(&hash.into_inner())
+                    .push_opcode(opcodes::all::OP_EQUALVERIFY)
+                    .push_opcode(opcodes::all::OP_CHECKSIG)
+                    .into_script();
```

Comment:
Making an address from a key should give you this script I think.

Reply:
Thanks for the tip! Funnily enough for a `p2wpkh` address you have to construct the corresponding `p2pkh` address and call `script_pubkey()` on it. Otherwise it doesn't work!